nevergrad/optimization/recastlib.py

```diff
@@ -63,21 +76,54 @@ def _optimization_function(self, objective_function: tp.Callable[[tp.ArrayLike],
         remaining: float = budget - self._num_ask
         while remaining > 0:  # try to restart if budget is not elapsed
             options: tp.Dict[str, tp.Any] = {} if self.budget is None else {"maxiter": remaining}
-            res = scipyoptimize.minimize(
-                objective_function,
-                best_x if not self.random_restart else self._rng.normal(0.0, 1.0, self.dimension),
-                method=self.method,
-                options=options,
-                tol=0,
-            )
-            if res.fun < best_res:
-                best_res = res.fun
-                best_x = res.x
+            if self.method == "CmaFmin2":
+                if p.helpers.Normalizer(self.parametrization).fully_bounded:
+                    # Tell CMA to work in [0, 1].
+                    options["bounds"] = [0.0, 1.0]
+
+                def cma_objective_function(data):
+                    # Hopefully the line below does nothing if unbounded and rescales from [0, 1] if bounded.
+                    if self._normalizer is not None:
+                        data = self._normalizer.backward(np.asarray(data, dtype=np.float32))
+                    return objective_function(data)
+
+                # cma.fmin2(objective_function, [0.0] * self.dimension, [1.0] * self.dimension, remaining)
+                x0 = [0.5] * self.dimension
+                num_calls = 0
+                while budget - num_calls > 0:
+                    options = {"maxfevals": budget - num_calls, "verbose": -9}
+                    res = cma.fmin(
+                        cma_objective_function,
+                        x0=x0,
+                        sigma0=0.2,
+                        options=options,
+                        restarts=9,
+                    )
+                    x0 = list(
+                        0.5 + np.random.uniform() * (2.0 * np.random.uniform(size=self.dimension) - 1) / 2.0
+                    )
```

Comment:
i dont get this

Reply:
Just randomly drawing a restart in the domain [0, 1]^d when bounded ?